### ISSUE TITLE
Must the number of categories be the same across training data when training for world detection? And I don't understand why the max sample num is set to min(nc,80).

### ISSUE THREAD

**Issue by imzhaoruijie** [ultralytics/ultralytics#16508]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Must the number of categories be the same across training data when training for world detection? 
When I use 2 data in trainnig, one with 365classes and another with 35classes, it can't work.
And I don't understand why the max sample num is set to min(nc,80).

**Comment by UltralyticsAssistant** [comment]
👋 Hello @imzhaoruijie, thank you for reaching out to Ultralytics 🚀! Your question about category consistency during training and the sample number limit is great.

For a 🐛 Bug Report, if applicable, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us investigate.

For custom training ❓ Questions, sharing more details like dataset examples, configurations, and logs will help. Please ensure you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Explore our community to learn and share:
- Join [Discord](https://ultralytics.com/discord) for real-time conversations 🎧.
- Engage in deeper discussions on [Discourse](https://community.ultralytics.com).
- Discover insights on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Ensure you're using the latest `ultralytics` package by running:

```bash
pip install -U ultralytics
```

This includes all necessary [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) for a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/).

## Environments

Run YOLOv8 in verified environments:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker**: [Docker Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If the badge is green, our [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. These are checked on macOS, Windows, and Ubuntu regularly.

This is an automated response, but an Ultralytics engineer will assist you soon.

**Comment by Skillnoob** [comment]
You can't use 2 different datasets while training unless you merge them together.
If you train a model on the dataset with 365 classes and then the same model on the dataset with 35, it will override the class count.
